### DISCUSSION TITLE
Segmentation fault on a tile

### DISCUSSION THREAD

**Discussion by Antguz** [davidfrantz/force#269] [Q&A] [answered]
Dear community,

I am trying to get LSP metrics on ARD, and there is a specific tile that gives me errors. The error says 'Segmentation fault' when I run `force-higher-level TSA`. The error does not appear when I run `force-higher-level CSO`. 

This is what I have done so far to explore the error on the tile:

- Re-run all the L1 scenes for that tile.
- Looking for potential .look files.
- And inspect each of the .tif files manually (this took me a while, and not sure if that was done properly).

I am running out of ideas, what can I do?

I am using FORCE 3.7.9 in a cluster through singularity. 

Thanks in advance.

AG.

**Comment by charalamm** [comment]
NOT an expert here. Just few ideas since nobody answered so far. Do you have enough disk space? Are the file and dir permissions ok?

**Comment by davidfrantz** [comment]
Hi @Antguz,

are you using the SPLITS or polar-based phenometrics?

Cheers,
David

**Reply by Antguz** [reply]
Hi @davidfrantz,

I am using polar-based phenometrics. I have a lot of space so that is no a problem and it happen in a single tile. If I skip that tile, the others runs well. Do you want me to share that tile with you?

AG.

**Reply by davidfrantz** [reply]
yes, that would certainly help!

**Reply by Antguz** [reply]
Sorry for the waiting @davidfrantz.  

Within this [link](https://drive.google.com/file/d/1equ-U9HkZ1vkwCLfPZG1AQz4kEMdjvEk/view?usp=share_link) (~30GB) you will be able to get two tiles for testing (ones that work and another one with the mentioned error). Also, this is the [.prm file](https://drive.google.com/file/d/1PMXGU3Z5IjjZIDQniv6xCTz1UtUOGsH2/view?usp=share_link) that I am using. 

Thanks in advance.

AG.

**Comment by davidfrantz** [comment, marked as answer]
Hi @Antguz,

I ran the parameter file on your data with debug-enabled FORCE and got reading errors on a couple of files (3x).

It turned out, that these had file sizes of 0. Not sure why that happened though. But I guess when you delete them, and run the preprocessing again, it will all be fine.

The segmentation fault was because the error handling was done a couple of lines too late in the reading routine. I already fixed this in the dev branch...

Cheers,
David
